light-node/src/commands/initialize.rs

```diff
@@ -0,0 +1,92 @@
+//! `intialize` subcommand
+
+use crate::application::app_config;
+use crate::config::LightClientConfig;
+
+use abscissa_core::{Command, Options, Runnable};
+use std::collections::HashMap;
+
+use tendermint::lite::Header;
+use tendermint::{hash, Hash};
+
+use tendermint_light_client::components::io::{AtHeight, Io, ProdIo};
+use tendermint_light_client::operations::ProdHasher;
+use tendermint_light_client::predicates::{ProdPredicates, VerificationPredicates};
+use tendermint_light_client::store::sled::SledStore;
+use tendermint_light_client::store::LightStore;
+use tendermint_light_client::types::Status;
+
+/// `intialize` subcommand
+#[derive(Command, Debug, Default, Options)]
+pub struct InitCmd {
+    #[options(
+        free,
+        help = "subjective height of the initial trusted state to initialize the node with"
+    )]
+    pub height: u64,
+
+    #[options(
+        free,
+        help = "hash of the initial subjectively trusted header to initialize the node with"
+    )]
+    pub header_hash: String,
+}
+
+impl Runnable for InitCmd {
+    fn run(&self) {
+        let subjective_header_hash =
+            Hash::from_hex_upper(hash::Algorithm::Sha256, &self.header_hash).unwrap();
+        let app_cfg = app_config();
+
+        let lc = app_cfg.light_clients.first().unwrap();
+
+        let mut peer_map = HashMap::new();
+        peer_map.insert(lc.peer_id, lc.address.clone());
+
+        let io = ProdIo::new(peer_map, Some(app_cfg.rpc_config.request_timeout));
+
+        initialize_subjectively(self.height, subjective_header_hash, &lc, &io);
+    }
+}
+
+fn initialize_subjectively(
+    height: u64,
+    subjective_header_hash: Hash,
+    l_conf: &LightClientConfig,
+    io: &ProdIo,
+) {
+    let db = sled::open(l_conf.db_path.clone()).unwrap_or_else(|e| {
+        println!("[ error ] could not open database: {}", e);
+        std::process::exit(1);
+    });
+
+    let mut light_store = SledStore::new(db);
+
+    if light_store.latest(Status::Verified).is_some() {
+        println!("[ warning ] overwriting trusted state in database");
+    }
+
+    let trusted_state = io
+        .fetch_light_block(l_conf.peer_id, AtHeight::At(height))
```

Comment:
Why we are doing IO outside light client (core)? Also all the checks that follow. I was expecting cli code to parse config file, do basic checks on the parameters obtained and then pass them to the light client init code. I am not sure even about db initialisation as we need to cover restarting scenario, and it feels more natural to let core library deals with this.

Reply:
What do you mean by light client (core)? Neither the [`Handle`](https://github.com/informalsystems/tendermint-rs/blob/b1252c639ece21aef5c6ac46f6e5b86187880480/light-client/src/supervisor.rs#L14-L34) nor the [`LightClient`](https://github.com/informalsystems/tendermint-rs/blob/b1252c639ece21aef5c6ac46f6e5b86187880480/light-client/src/light_client.rs#L51) expose a method for intialization. Maybe I'm missing sth here? cc @romac @brapse

I agree that this should live in the light-client crate instead. There is a related discussion around what should live in the core library vs here:  
- https://github.com/informalsystems/tendermint-rs/pull/430#discussion_r452662130
- https://github.com/informalsystems/tendermint-rs/pull/430#discussion_r452656444